Shipping Labels: Only show Add Payment Button if current user is store owner

hafizrahman opened the pull request:
Fixes #4321

This PR hides the add payment button if the current user is not store owner, and the payment methods list is not empty.

To test:

1. Have a test site with Shipping labels feature enabled, payment method already added, and the site already having an account in it that is not the store owner,
2. Log in to app as a non-store owner account,
3. Find an Order and create a shipping label,
4. Go through all steps until the "Payment method" step is active.
5. Tap "Edit" on the "Payment method" step.
6. Make sure "Add another credit card" button is not shown.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

hafizrahman committed:
Only show add payment btn if is store owner and paymnt methods not empty

hichamboushaba reviewed (changes requested):
Thanks @hafizrahman for working on this.
But I found two issues while testing this:
1. The add button is displayed if there are no payment methods available
<img width=360 src="https://user-images.githubusercontent.com/1657201/127169280-4c8f96f2-5387-47ca-952f-7c7a7de15203.png"/>
2. This wasn't specified in the original ticket, and it's not related to your work, but related to the issue: if you don't have any payment methods, and you open the form with a non-owner account, when you get to the payment step, the app will take you to add a payment method automatically.

For the issue 1, I suggest you remove the `canShowAddPaymentButton` variable, as it's not needed, and base your logic upon `canManagePayments` in two places:
[here](https://github.com/woocommerce/woocommerce-android/blob/e19a9c0b1d810135e4f094557e1302c9164e5d55/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelPaymentFragment.kt#L135) with something like
```kotlin
new.canManagePayments.takeIfNotEqualTo(old?.canManagePayments) { canManagePayments ->
                binding.editWarningBanner.isVisible = !canManagePayments
                paymentMethodsAdapter.isEditingEnabled = canManagePayments
                binding.paymentMethodsSectionTitle.isEnabled = canManagePayments
                binding.addPaymentMethodButton.isVisible = canManagePayments && new.paymentMethods.isNotEmpty()
                binding.addFirstPaymentMethodButton.isVisible = canManagePayments && new.paymentMethods.isEmpty()
            }
```
and [here](https://github.com/woocommerce/woocommerce-android/blob/e19a9c0b1d810135e4f094557e1302c9164e5d55/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelPaymentFragment.kt#L139) by changing it to something like this:
```kotlin
it.isEmpty().let { isListEmpty ->
                    binding.paymentMethodsSectionTitle.isVisible = !isListEmpty
                    binding.paymentMethodsList.isVisible = !isListEmpty
                    binding.addPaymentMethodButton.isVisible = !isListEmpty && new.canManagePayments
                    binding.addFirstPaymentMethodButton.isVisible = isListEmpty && new.canManagePayments
                }
```

For the second issue, I think as a short term fix, we can just avoid redirecting the user to the web view if `canManagerPayments` is false (we should update the logic [here](https://github.com/woocommerce/woocommerce-android/blob/25aad10cf58f3f8051be4b4108cabf7ad51281d3/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelPaymentViewModel.kt#L37)), this way the user will understand why they can't continue the shipping label creation process. And the ideal fix would be to adjust the plugin to mark this case as `non-eligible` for shipping label creation, I'll try to take care of this one.

Let me know if something is not clear 🙂

hafizrahman committed:
Handle "Add First Payment" button case, too.

hafizrahman committed:
Don't redirect to add payment method for non-store owner.

This is a short term fix, see here for more details: https://github.com/woocommerce/woocommerce-android/pull/4493#pullrequestreview-715986558

hichamboushaba reviewed (approved):
Looks good, thanks for taking care of this Hafiz.
Do you think we can add some tests to cover the cases of this PR?

I'm pre-approving 🙂

hafizrahman committed:
Unit tests related to non-store owner handling in edit payment method.

hafizrahman commented:
Good reminder, unit tests added @hichamboushaba

hichamboushaba commented:
Thanks @hafizrahman :shipit:
